src/libLookup/Lookup.cpp

```diff
@@ -3553,14 +3553,22 @@ bool Lookup::ProcessGetDSGuardNetworkInfo(const bytes& message,
 
   uint32_t portNo = 0;
   uint64_t dsEpochNo = 0;
+  PubKey senderPubKey;
 
   if (!Messenger::GetLookupGetNewDSGuardNetworkInfoFromLookup(
-          message, offset, portNo, dsEpochNo)) {
+          message, offset, portNo, dsEpochNo, senderPubKey)) {
     LOG_EPOCH(WARNING, m_mediator.m_currentEpochNum,
               "Messenger::GetLookupGetNewDSGuardNetworkInfoFromLookup failed.");
     return false;
   }
 
+  if (!IsLookupNode(senderPubKey)) {
```

Comment:
The sender of the message is actually from shard node -> lookup. So this check is not correct.